projects/dotcms-field-elements/src/components/dot-key-value/dot-key-value.e2e.ts

```diff
@@ -0,0 +1,166 @@
+
+        it('should have required as false', async () => {
+            element.setProperty('required', 'false');
+            await page.waitForChanges();
+            const required = await element.getProperty('required');
+            const label = await element.find('.dot-field__label label');
+            expect(required).toBeFalsy();
+            expect(label.innerHTML.indexOf('*')).toBe(-1);
+        });
+
+        describe('Events', () => {
+            beforeEach(async () => {
+                spyStatusChangeEvent = await page.spyOnEvent('statusChange');
+                spyValueChangeEvent = await page.spyOnEvent('valueChange');
+            });
+
+            it('should emit "statusChange" & "valueChange" and add the new row in "Key-Value-Table" when saving an item', async () => {
+                const inputs = await page.findAll('input');
+                const saveBtn = await page.find('.dot-key-value__save__button');
+                await inputs[0].press('k');
+                await inputs[1].press('2');
+                saveBtn.click();
+                await page.waitForChanges();
+
+                const tableRows = await page.findAll('table tr');
+                expect(tableRows.length).toBe(2);
+                expect(spyStatusChangeEvent).toHaveReceivedEventDetail({
+                    name: 'testName',
+                    status: {
+                        dotPristine: false,
+                        dotTouched: true,
+                        dotValid: true
+                    }
+                });
+                expect(spyValueChangeEvent).toHaveReceivedEventDetail({
+                    name: 'testName',
+                    fieldType: 'Key-Value',
+                    value: 'valueA|1,k|2'
+                });
+            });
+
+            it('should emit status and value on Reset', async () => {
+                element.callMethod('reset');
+                await page.waitForChanges();
+                expect(spyStatusChangeEvent).toHaveReceivedEventDetail({
+                    name: 'testName',
+                    status: {
+                        dotPristine: true,
+                        dotTouched: false,
+                        dotValid: false
+                    }
+                });
+                expect(spyValueChangeEvent).toHaveReceivedEventDetail({
+                    name: 'testName',
+                    fieldType: 'Key-Value',
+                    value: ''
+                });
+            });
+
+            it('should emit status and value on "deleteItemEvt" and remove the row in "Key-Value-Table" event', async () => {
+                await element.triggerEvent('deleteItemEvt', {
+                    bubbles: true,
+                    cancelable: false,
+                    detail: 0
+                });
+                await page.waitForChanges();
+
+                const tableRows = await page.findAll('table tr');
+                expect(tableRows.length).toBeFalsy();
```

Comment:
```suggestion
                expect(tableRows.length).toBe(0);
```

Reply:
when there are no rows, the `tableRows` find selector, will return `null`, because no markup is rendered. So I cannot compare it to `0`